src/main/java/org/skypro/skyshop/model/service/SearchService.java

```diff
@@ -0,0 +1,29 @@
+package org.skypro.skyshop.model.service;
+
+import org.skypro.skyshop.model.search.SearchResult;
+import org.skypro.skyshop.model.search.Searchable;
+import org.springframework.stereotype.Service;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class SearchService {
+    private final StorageService storageService;
+
+    public SearchService(StorageService storageService) {
+        this.storageService = storageService;
+    }
+
+    public List<SearchResult> search(String pattern) {
+        String lowerPattern = pattern.toLowerCase();
+
+        Collection<Searchable> allItems = storageService.getAllSearchables();
+
+        return allItems.stream()
+                .filter(item -> item.getName().toLowerCase().contains(lowerPattern))
+                .map(item -> new SearchResult(item.getId(), item.getName(), item.getContentType()))
+                .collect(Collectors.toList());
```

Comment:
можно сократить код до одной строки

```suggestion
        return storageService.getAllSearchables().stream()
                .filter(item -> item.getName().toLowerCase().contains(pattern.toLowerCase()))
                .map(item -> new SearchResult(item.getId(), item.getName(), item.getContentType()))
                .collect(Collectors.toList());
```